### PR TITLE
compatibility with ocaml PR#1569 (changes to compiler install dir)

### PR DESCRIPTION
**Pull request by sliquister**
See description of https://github.com/ocaml/ocaml/pull/1569 .

### CODE REVIEW

File: src/meta.ml [hunk 1]
```diff
@@ -207,24 +212,29 @@ let builtins ~stdlib_dir =
...
         [ requires []
         ; version
-        ; directory "+compiler-libs"
-        ; sub "common" []
-        ; sub "bytecomp" ["compiler-libs.common"  ]
-        ; sub "optcomp"  ["compiler-libs.common"  ]
+        ]
+        @ (if sep_libraries then [] else [ directory "+compiler-libs" ])
+        @
+        [ sub "common" []
```
Comment (diml) [inline src/meta.ml]: ocamlcommon is not in a sub-directory as well?

File: src/meta.ml [hunk 2]
```diff
@@ -195,6 +195,11 @@ let archives name =
 
 let builtins ~stdlib_dir =
   let version = version "[distributed with Ocaml]" in
+  let sep_libraries =
+    List.exists (fun suf ->
```
Comment (diml) [inline src/meta.ml]: the label is missing (that's what causing the CI failures)

File: src/meta.ml (hunk 1 above)
Comment (sliquister) [reply src/meta.ml]: It is, I guessed I missed this change and for some reason, it didn't break anything.

File: src/meta.ml (hunk 2 above)
Comment (sliquister) [reply src/meta.ml]: Hum, weird I was not getting this earlier but instead I had warning 32 in usexp.mli. Not sure what happened but I fixed this.

File: src/meta.ml (hunk 1 above)
Comment (sliquister) [reply src/meta.ml]: Fixed now.

**Comment by diml** [comment]
Thanks. We'll merge this PR as soon as the OCaml PR is merged

**Pull request closed without merging**